test/e2e/benchmark/benchmark.go

```diff
@@ -118,3 +121,56 @@ func (b *BenchmarkTest) Run() error {
 
 	return nil
 }
+
+func (b *BenchmarkTest) CheckResults(expectedBlockSizeBytes int64) error {
+	log.Println("Checking results")
+
+	// if local tracing was enabled,
+	// pull block summary table from one of the nodes to confirm tracing
+	// has worked properly.
+	if b.manifest.LocalTracingType == "local" {
+		if _, err := b.Node(0).PullBlockSummaryTraces("."); err != nil {
+			return fmt.Errorf("failed to pull traces: %w", err)
+		}
+	}
+
+	// download traces from S3, if enabled
+	if b.manifest.PushTrace && b.manifest.DownloadTraces {
+		// download traces from S3
+		pushConfig, err := trace.GetPushConfigFromEnv()
+		if err != nil {
+			return fmt.Errorf("failed to get push config: %w", err)
+		}
+		err = trace.S3Download("./traces/", b.manifest.ChainID,
+			pushConfig)
+		if err != nil {
+			return fmt.Errorf("failed to download traces from S3: %w", err)
+		}
+	}
+
+	log.Println("Reading blockchain")
+	blockchain, err := testnode.ReadBlockchain(context.Background(),
+		b.Node(0).AddressRPC())
+	testnet.NoError("failed to read blockchain", err)
+
+	targetSizeReached := false
+	maxBlockSize := int64(0)
+	for _, block := range blockchain {
+		if appconsts.LatestVersion != block.Version.App {
+			return fmt.Errorf("expected app version %d, got %d", appconsts.LatestVersion, block.Version.App)
+		}
+		size := int64(block.Size())
+		if size >= expectedBlockSizeBytes {
+			targetSizeReached = true
+			break
+		}
+		if size > maxBlockSize {
+			maxBlockSize = size
+		}
```

Comment:
nit: I don't think these lines will be executed if the previous conditional evaluates to `true` b/c the `break` so consider re-ordering this above the previous conditional so that `maxBlockSize` accounts for the block that actually exceeds >= expectedBlockSizeBytes

Reply:
The current version is correct as well. I see what causes confusion here: the value of `maxBlockSize` is only used when the target size is not reached. That is, `if size >= expectedBlockSizeBytes` is never hit, and if that condition is never met, then `maxBlockSize` correctly reflects the most recent maximum size observed. If we do hit the `expectedBlockSizeBytes` condition and `if size >= expectedBlockSizeBytes` evaluates to true, then the value of `maxBlockSize` does not matter. Nevertheless, to resolve future confusion, I will apply your suggestion in a follow up PR, thanks!

Reply:
Please see https://github.com/celestiaorg/celestia-app/pull/3670